### ISSUE TITLE
Socket Reconnection

### ISSUE THREAD

**Issue by yjovian** [nolanlawson/socket-pouch#3]
Doesn't reconnect if socket dropped.  Could implement socket.io client manager reconnection.  Quick workaround in browser in a ten second timer function:

retryReplication();

function retryReplication(){
if (remoteDB._socket.readyState=="closed"){
        remoteDB._close();
        remoteDB = new PouchDB(opts);
        localSync.cancel();
        localSync = localDB.sync(remoteDB, {live: true, retry:true})
}
setTimeout(retryReplication,10000);
}

**Comment by nolanlawson** [comment]
PouchDB already has the `retry` option, which ought to work correctly if this library is throwing errors where it should. :)

In general, though, you should consider `socket-pouch` to be highly experimental. Not production ready!

**Comment by yjovian** [comment]
Yes, retry works if I lose my internet connection, but if I restart my server (which I do a lot right now) it doesn't reconnect because I guess there's nothing to tie back into on the server side.  Socket.io, however, does reconnect even after server restarts.  It must be storing it's id's somewhere.  I'll see if I can duplicate that functionality on the server.  

As far as production status, I understand, but it is working flawlessly for me in every browser (ie,ff,chrome,safari) and on ipad, nexus7, and my htc sensation.  Very nice piece of work, Nolan.  I'll look forward to an updated version, and i'll keep using/testing and let you know if I find anything else.

**Comment by nolanlawson** [comment]
You could also submit a pull request. :)

**Comment by cellog** [comment]
This bug is a huge thing.  At the very least document it prominently in the README, because this bug causes a silent fail, even with logging on. I had a very serious real-world mess up today because some data on the server never synced to a client.
